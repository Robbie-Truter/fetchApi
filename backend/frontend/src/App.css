.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: rgba(0, 0, 0, 0);
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: #000000;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

button{
  width: 17%;
}


img{
  height:180px;
}


input {
  padding: 12px 20px;
  margin: 8px 0;
  box-sizing: border-box;
  border: none;
  background-color: #56e0c0;
  color: white;
}


p{
  font-size:10px;
}