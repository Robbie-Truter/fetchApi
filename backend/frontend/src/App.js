import "./App.css";
import ColoredLine from "./ColoredLine";
import React from "react";


class App extends React.Component {
    constructor(props) {
        super(props);
        //states for storing each api result
        this.state = {
            apiResponse1: "" ,
            apiResponse2: "",
            apiResponse3: "",
            value: "",
        };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.setState({
            value: event.target.value
        })
    };

    // Triggers fetch request when submit is clicked
    handleSubmit(event) {
        //fetch each third party api from backend
        //storing api data into empty states
        fetch("http://localhost:8000/github/" + this.state.value)
            .then(res => res.text())
            .then(res => this.setState({apiResponse1: res}))
            .catch(err => err);


        //fetch gitLab api from backend
        fetch("http://localhost:8000/gitlab/" + this.state.value)
            .then(res => res.text())
            .then(res => this.setState({apiResponse2: res}))
            .catch(err => err);


        //fetch bitBucket api from backend
        fetch("http://localhost:8000/bitbucket/" + this.state.value)
            .then(res => res.text())
            .then(res => this.setState({apiResponse3: res}))
            .catch(err => err);

        event.preventDefault();
    }


    render() {
        return (
                <header className="App-header">

                    <div className="App">
                    <h2><b>Third party api user details:</b></h2>
                        <form onSubmit={this.handleSubmit}>
                        <label>
                            <input type="text" placeholder={"Enter username here.."} value={this.state.value} onChange={this.handleChange} />
                            <br />
                            <input type="submit" value="Submit" />
                        </label>
                    </form>

                        <ColoredLine />
                        <img src={"/gitHub.gif"} alt="gitHub logo" />
                        <p>{this.state.apiResponse1}</p>

                        <ColoredLine />
                        <img src={"/gitLab.gif"} alt="gitLab logo" />
                        <p>{this.state.apiResponse2}</p>

                        <ColoredLine />
                        <img src={"/bitbucket.gif"} alt="Bitbucket logo" />
                        <p><b>(Note: Have to enter valid user id for Bitbucket results)</b></p>
                        <p>{this.state.apiResponse3}</p>
                    </div>

                </header>
        );
    }
}


export default App;