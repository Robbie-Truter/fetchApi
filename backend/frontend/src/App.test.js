import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/(Note: Have to enter valid user id for Bitbucket results)/i);
  expect(linkElement).toBeInTheDocument();
});


it('renders correctly', () => {
  const tree = renderer
      .create(<App page="http://localhost:8000/github/Timothylang-tech">Instagram</App>)
      .toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders correctly', () => {
  const tree = renderer
      .create(<App page="http://localhost:8000/gitlab/test">Instagram</App>)
      .toJSON();
  expect(tree).toMatchSnapshot();
});


