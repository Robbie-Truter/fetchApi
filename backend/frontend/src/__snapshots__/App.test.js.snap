// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="App-header"
>
  <div
    className="App"
  >
    <h2>
      <b>
        Third party api user details:
      </b>
    </h2>
    <form>
      <label>
        <input
          onChange={[Function]}
          placeholder="Enter username here.."
          type="text"
          value=""
        />
      </label>
    </form>
    <hr
      style={
        Object {
          "height": 0.5,
          "opacity": 0.5,
        }
      }
    />
    <img
      alt="gitHub logo"
      src="/gitHub.gif"
    />
    <p>
      
    </p>
    <hr
      style={
        Object {
          "height": 0.5,
          "opacity": 0.5,
        }
      }
    />
    <img
      alt="gitLab logo"
      src="/gitLab.gif"
    />
    <p>
      
    </p>
    <hr
      style={
        Object {
          "height": 0.5,
          "opacity": 0.5,
        }
      }
    />
    <img
      alt="Bitbucket logo"
      src="/bitbucket.gif"
    />
    <p>
      <b>
        (Note: Have to enter valid user id for Bitbucket results)
      </b>
    </p>
    <p>
      
    </p>
  </div>
</header>
`;

exports[`renders correctly 2`] = `
<header
  className="App-header"
>
  <div
    className="App"
  >
    <h2>
      <b>
        Third party api user details:
      </b>
    </h2>
    <form>
      <label>
        <input
          onChange={[Function]}
          placeholder="Enter username here.."
          type="text"
          value=""
        />
      </label>
    </form>
    <hr
      style={
        Object {
          "height": 0.5,
          "opacity": 0.5,
        }
      }
    />
    <img
      alt="gitHub logo"
      src="/gitHub.gif"
    />
    <p>
      
    </p>
    <hr
      style={
        Object {
          "height": 0.5,
          "opacity": 0.5,
        }
      }
    />
    <img
      alt="gitLab logo"
      src="/gitLab.gif"
    />
    <p>
      
    </p>
    <hr
      style={
        Object {
          "height": 0.5,
          "opacity": 0.5,
        }
      }
    />
    <img
      alt="Bitbucket logo"
      src="/bitbucket.gif"
    />
    <p>
      <b>
        (Note: Have to enter valid user id for Bitbucket results)
      </b>
    </p>
    <p>
      
    </p>
  </div>
</header>
`;
